src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4095,6 +4446,20 @@ public override BoundNode VisitMethodGroup(BoundMethodGroup node)
 
         public override BoundNode VisitLambda(BoundLambda node)
         {
+            // It's possible to reach VisitLambda without having analyzed the lambda body in error scenarios,
+            // so we analyze for the purposes of determining top-level nullability. We don't want to report
+            // any diagnostics from this analysis, as scenarios we want to have diagnostics for will have had
+            // them reported through other analysis steps.
+
+            // PROTOTYPE(nullable-api): Can we make this conditional on whether or not we've already seen the node
+            // or will that have no effect because this is always called first? It is for at least one lambda
+            // conversion case, need to investigate others
+            if (!_disableNullabilityAnalysis)
+            {
+                var bag = new DiagnosticBag();
+                Analyze(compilation, node, bag, node.Type.GetDelegateType()?.DelegateInvokeMethod, returnTypes: null, initialState: GetVariableState(), _analyzedNullabilityMapOpt);
```

Comment:
>Analyze [](start = 16, length = 7)

Are we visiting the lambda body twice in non-error scenarios?

Reply:
Yes. The prototype comment above is tracking trying to remove this.

---
In reply to: [266115689](https://github.com/dotnet/roslyn/pull/34017#discussion_r266115689) [](ancestors = 266115689)